src/nunit.analyzers/DisposeFieldsAndPropertiesInTearDown/DisposeFieldsAndPropertiesInTearDownAnalyzer.cs

```diff
@@ -541,6 +544,22 @@ public Parameters(SemanticModel model, INamedTypeSymbol type, ImmutableHashSet<s
                     SymbolEqualityComparer.Default.Equals(calledMethod.ContainingType, this.type);
             }
 
+            public void ResetMethodCallVisits() => this.visitedMethods.Clear();
+
+            public bool IsLocalNotYetVisitedMethodCall(
+                InvocationExpressionSyntax invocationExpression,
+                [NotNullWhen(true)] out IMethodSymbol? calledMethod)
+            {
+                if (this.IsLocalMethodCall(invocationExpression, out calledMethod) &&
```

Comment:
The whole method simplifies to
```
return this.IsLocalMethodCall(invocationExpression, out calledMethod) &&
       this.visitedMethods.Add(calledMethod);
```

Reply:
Thanks, excellent suggestion. Code updated.